crates/sargon/src/gateway_api/methods/state_methods.rs

```diff
@@ -635,3 +699,133 @@ mod fetch_all_resources_tests {
         ))
     }
 }
+
+#[cfg(test)]
+mod filter_transferable_tests {
+    use crate::prelude::*;
+    use actix_rt::time::timeout;
+
+    #[allow(clippy::upper_case_acronyms)]
+    type SUT = GatewayClient;
+
+    #[actix_rt::test]
+    async fn filter_resources() {
+        // Test the case where all resources are transferable
+
+        // Mock the resources
+        let fungible_transferable = FungibleResourcesCollectionItem::sample();
+        let fungible_not_transferable =
+            FungibleResourcesCollectionItem::sample_other();
+        let non_fungible_transferable =
+            NonFungibleResourcesCollectionItem::sample();
+        let non_fungible_not_transferable =
+            NonFungibleResourcesCollectionItem::sample_other();
+        let output = FetchResourcesOutput::new(
```

Comment:
If `FetchResourcesOutput`'s ctor `new` used `impl IntoIterator<Item =>` as params the `vec!`s would not be needed :)